cmd/build.go

```diff
@@ -91,5 +60,105 @@ func Build(ctx context.Context) *cobra.Command {
 	cmd.Flags().StringVarP(&options.OutputMode, "progress", "", "tty", "show plain/tty build output")
 	cmd.Flags().StringArrayVar(&options.BuildArgs, "build-arg", nil, "set build-time variables")
 	cmd.Flags().StringArrayVar(&options.Secrets, "secret", nil, "secret files exposed to the build. Format: id=mysecret,src=/local/secret")
+	cmd.Flags().StringVarP(&options.Namespace, "namespace", "", "", "namespace against which the image will be consumed. Default is the one defined at okteto context or okteto manifest")
 	return cmd
 }
+
+func isManifestV2(file string) (model.ManifestBuild, bool) {
+	r, err := strconv.ParseBool(os.Getenv("OKTETO_ENABLE_MANIFEST_V2"))
+	if err != nil || !r {
+		return nil, false
+	}
+	mPath := contextCMD.GetOktetoManifestPath(file)
+	if mPath == "" {
+		return nil, false
+	}
+
+	mBuild, err := model.GetBuildManifest(mPath)
+	if err != nil || mBuild == nil {
+		return nil, false
+	}
+	return mBuild, true
+}
+
+func buildV2(m model.ManifestBuild, options build.BuildOptions, args []string) error {
+	service := ""
+	if len(args) == 1 {
+		service = args[0]
+	}
+
+	if service != "" {
+		b, ok := m[service]
+		if !ok {
+			return fmt.Errorf("invalid service name")
+		}
+
+		if options.Target != "" {
+			b.Target = options.Target
+		}
+		if len(options.CacheFrom) != 0 {
+			b.CacheFrom = options.CacheFrom
+		}
+
+		opts := build.OptsFromManifest(service, b)
+		opts.Secrets = options.Secrets
+
+		err := buildV1(opts, []string{service})
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	for service, b := range m {
+		opts := build.OptsFromManifest(service, b)
+		err := buildV1(opts, []string{service})
```

Comment:
Aren't we assuming that the service name would be the same than the folder name? Checking the `buildV1`, service would be used as a path. Shouldn't we use the `context` defined in the manifest for that specific service or `.` by default? the same for line 106 in this file.

I mean, If I have this manifest:

```
...
build:
  api:
    context: backend
    dockerfile: Dockerfile
    ....
...
```

and I execute `okteto build api`, aren't we passing `api` as `context` instead of `backend` to the BuildV1 function?